hudi-spark/src/test/java/org/apache/hudi/keygen/TestTimestampBasedKeyGenerator.java

```diff
@@ -107,6 +107,13 @@ public void testTimestampBasedKeyGenerator() throws IOException {
     properties = getBaseKeyConfig("DATE_STRING", "yyyy-MM-dd hh", "GMT", null);
     HoodieKey hk4 = new TimestampBasedKeyGenerator(properties).getKey(baseRecord);
     assertEquals("2020-01-06 12", hk4.getPartitionPath());
+
+    // timestamp is DATE_STRING, timezone is GMT+8:00, createTime is null
+    baseRecord.put("createTime", null);
+    properties = getBaseKeyConfig("DATE_STRING", "yyyy-MM-dd hh:mm:ss", "GMT+8:00", null);
```

Comment:
Let us add a test case for the scenario when timestampType is SCALAR or EPOCH_MILLISECONDS and the value of the field is null?

Reply:
> Let us add a test case for the scenario when timestampType is SCALAR or EPOCH_MILLISECONDS and the value of the field is null?

done